applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -335,41 +331,33 @@ void SmallStrainUPwDiffOrderElement::
     const SizeType NumUNodes = rGeom.PointsNumber();
     const SizeType NumPNodes = mpPressureGeometry->PointsNumber();
 
-    // const bool IgnoreUndrained = GetProperties()[IGNORE_UNDRAINED];
-    // const SizeType ElementSize = (IgnoreUndrained ? NumUNodes * Dim : NumUNodes * Dim + NumPNodes);
     const SizeType ElementSize = NumUNodes * Dim + NumPNodes;
 
     if (rElementalDofList.size() != ElementSize)
         rElementalDofList.resize(ElementSize);
 
     SizeType Index = 0;
-
-    if (Dim > 2) {
-        for (SizeType i = 0; i < NumUNodes; ++i) {
-            rElementalDofList[Index++] = GetGeometry()[i].pGetDof( DISPLACEMENT_X );
-            rElementalDofList[Index++] = GetGeometry()[i].pGetDof( DISPLACEMENT_Y );
-            rElementalDofList[Index++] = GetGeometry()[i].pGetDof( DISPLACEMENT_Z );
-        }
-    } else {
-        for (SizeType i = 0; i < NumUNodes; ++i) {
-            rElementalDofList[Index++] = GetGeometry()[i].pGetDof( DISPLACEMENT_X );
-            rElementalDofList[Index++] = GetGeometry()[i].pGetDof( DISPLACEMENT_Y );
+    for (SizeType i = 0; i < NumUNodes; ++i) {
+        rElementalDofList[Index] = GetGeometry()[i].pGetDof( DISPLACEMENT_X );
+        ++Index;
+        rElementalDofList[Index] = GetGeometry()[i].pGetDof( DISPLACEMENT_Y );
+        ++Index;
+        if (Dim > 2) {
+            rElementalDofList[Index] = GetGeometry()[i].pGetDof(DISPLACEMENT_Z);
+            ++Index;
         }
     }
-
-    // if (!IgnoreUndrained) {
-        for (SizeType i=0; i < NumPNodes; ++i)
-            rElementalDofList[Index++] = GetGeometry()[i].pGetDof( WATER_PRESSURE );
-    // }
-
-    KRATOS_CATCH( "" )
+    for (SizeType i=0; i < NumPNodes; ++i) {
+        rElementalDofList[Index] = GetGeometry()[i].pGetDof(WATER_PRESSURE);
+        ++Index;
+    }
```

Comment:
Nice cleanup here, are there tests to cover this by any chance? Especially with these 'Index++', things can be overlooked quite quickly